hunts/src/puzzle-table.js

```diff
@@ -117,6 +123,7 @@ export const PuzzleTable = React.memo(
           <tbody {...getTableBodyProps()}>
             {rows.map((row, i) => {
               prepareRow(row);
+              row.original.toggleTagInFilter = toggleTagInFilter;
```

Comment:
Hmm I don't understand the data flow in this PR yet but I think we should generally avoid mutating row.original slash the row objects if possible. Is this just attached so that it's available in the right places? If so I think it's better passed around as a prop or stored in redux

Reply:
Ok, yeah, looking at it more closely I think this is just so it's available inside `TagCell`. I think we should use redux instead; it'll be a little more boilerplate but overall a bit cleaner (this kind of thing was exactly why I added redux to this project to begin with, otherwise managing the flow of all these things through the component tree gets kinda messy).

Reply:
This sounds like a good idea but I'm not sure how to do it. I think it involves doing something like in modalSlice.js?

Reply:
Yep, something like that. Probably `filterSlice` or `tagFilterSlice` or something. You'll be specifying something that has some state (initially `initialState`, for you probably an empty list) and some `reducers` (probably like `addTag` and `removeTag`) which can handle actions dispatched elsewhere and compute the new state. I think the modal example is pretty similar to what you want, though you don't have to worry about `extraReducers` I don't think. Let me know if that makes sense; it took me a while to internalize all of redux's abstraction layers from reading their documentation but hopefully the modal code is simple enough to follow.